File: Backend/renderer/src/main/java/net/chrisparton/sparkled/renderer/util/ColorUtils.java
package net.chrisparton.sparkled.renderer.util;

import java.awt.*;

public class ColorUtils {

    private ColorUtils() {
    }

    public static Color adjustBrightness(Color color, float brightness) {
        float[] hsb = Color.RGBtoHSB(color.getRed(), color.getGreen(), color.getBlue(), null);
        hsb[2] *= brightness;
        return new Color(Color.HSBtoRGB(hsb[0], hsb[1], hsb[2]));
    }
}

File: renderer/src/main/kotlin/io/sparkled/renderer/util/ColorUtils.kt
package io.sparkled.renderer.util

import java.awt.Color

object ColorUtils {

    fun adjustBrightness(color: Color, brightness: Float): Color {
        val hsb = Color.RGBtoHSB(color.red, color.green, color.blue, null)
        return Color(Color.HSBtoRGB(hsb[0], hsb[1], hsb[2] * brightness))
    }
}
